### ISSUE TITLE
swtpm-libs fails to install on jammy

### ISSUE THREAD

**Issue by popey** [quickemu-project/quickemu#448]
#### Expected behaviour

Upgraded to jammy and re-enabled the PPA post-upgrade. Did a dist-upgrade to get latest quickemu. 

Expected it to update/install okay.

#### Actual behaviour

One of the dependencies from the ppa failed to install correctly.

```
Preparing to unpack .../12-swtpm-libs_0.6.0-1~jammy1.2_amd64.deb ...
Unpacking swtpm-libs:amd64 (0.6.0-1~jammy1.2) over (0.6.0-1~impish1.2) ...
dpkg: error processing archive /tmp/apt-dpkg-install-ye4OHS/12-swtpm-libs_0.6.0-1~jammy1.2_amd64.deb (--unpack):
 trying to overwrite '/usr/lib/x86_64-linux-gnu/swtpm/libswtpm_libtpms.so.0.0.0', which is also in package swtpm 0.6.3-0ubuntu3
Errors were encountered while processing:
 /tmp/apt-dpkg-install-ye4OHS/12-swtpm-libs_0.6.0-1~jammy1.2_amd64.deb
E: Sub-process /usr/bin/dpkg returned an error code (1)
```

#### Steps to reproduce the behaviour

1. Upgrade Kubuntu from Impish to Jammy
2. Re-enable the quickemu ppa
3. Update packages

#### Quickemu output

```
quickemu --version
3.15
```
#### Linux Distribution & Kernel

```bash
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 22.04 LTS
Release:        22.04
Codename:       jammy
Linux robot 5.15.0-27-generic #28-Ubuntu SMP Thu Apr 14 04:55:28 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux
```

**Comment by popey** [comment]
I removed and reinstalled swtpm to get it to pull in the right version.

```
$ sudo apt install quickemu
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
The following packages were automatically installed and are no longer required:
  intel-cmt-cat kdevelop57-libs libbson-1.0-0 libdbi1 libesmtp6 libgdbm-compat4:i386 libgdbm6:i386 libgrpc++1 libgrpc10 libmemcached11 libmicrohttpd12 libmodbus5 libmongoc-1.0-0 libmongocrypt0 libmosquitto1 libopenipmi0 liboping0
  libow-3.2-4 libowcapi-3.2-4 libprotobuf-c1 libpython2-dev libpython2.7 libpython2.7-dev libqpid-proton11 librdkafka1 libriemann-client0 librrd8 libvarnishapi2 libvirt0 libxencall1 libxendevicemodel1 libxenevtchn1
  libxenforeignmemory1 libxengnttab1 libxentoolcore1 libxentoollog1 owfs-common python2-dev python2.7-dev rrdtool ruby-minitest ruby-power-assert ruby-test-unit swtpm-libs
Use 'sudo apt autoremove' to remove them.
The following additional packages will be installed:
  swtpm
The following NEW packages will be installed
  quickemu swtpm
0 to upgrade, 2 to newly install, 0 to remove and 1 not to upgrade.
Need to get 47.5 kB/79.6 kB of archives.
After this operation, 284 kB of additional disk space will be used.
Do you want to continue? [Y/n] 
Get:1 http://gb.archive.ubuntu.com/ubuntu jammy/main amd64 swtpm amd64 0.6.3-0ubuntu3 [47.5 kB]
Fetched 47.5 kB in 0s (465 kB/s)
Selecting previously unselected package swtpm.
(Reading database ... 507045 files and directories currently installed.)
Preparing to unpack .../swtpm_0.6.3-0ubuntu3_amd64.deb ...
Unpacking swtpm (0.6.3-0ubuntu3) ...
Selecting previously unselected package quickemu.
Preparing to unpack .../quickemu_3.15-1~jammy1.0_all.deb ...
Unpacking quickemu (3.15-1~jammy1.0) ...
Setting up swtpm (0.6.3-0ubuntu3) ...
Setting up quickemu (3.15-1~jammy1.0) ...
Processing triggers for man-db (2.10.2-1) ...
Processing triggers for libc-bin (2.35-0ubuntu3) ...
/sbin/ldconfig.real: /lib/libndi.so.4 is not a symbolic link

```

**Comment by popey** [comment]
This keeps happening. Every time I `deb-get upgrade` it makes my swtpm libs go wonky.

```
$ sudo deb-get upgrade
[sudo] password for alan: 
Reading package lists...
Building dependency tree...
Reading state information...
Calculating upgrade...
The following packages were automatically installed and are no longer required:
  intel-cmt-cat kdevelop57-libs libbson-1.0-0 libcmark-gfm-extensions0.29.0.gfm.3 libcmark-gfm0.29.0.gfm.3 libdbi1 libesmtp6 libgdbm-compat4:i386 libgdbm6:i386 libgrpc++1 libgrpc10 libmemcached11 libmicrohttpd12 libmodbus5
  libmongoc-1.0-0 libmongocrypt0 libmosquitto1 libopenipmi0 liboping0 libow-3.2-4 libowcapi-3.2-4 libprotobuf-c1 libpython2-dev libpython2.7 libpython2.7-dev libqpid-proton11 librdkafka1 libriemann-client0 librrd8 libsdl-ttf2.0-0
  libvarnishapi2 libvirt0 libxencall1 libxendevicemodel1 libxenevtchn1 libxenforeignmemory1 libxengnttab1 libxentoolcore1 libxentoollog1 owfs-common pandoc-data python-is-python2 python2-dev python2.7-dev rrdtool ruby-minitest
  ruby-power-assert ruby-test-unit swtpm-libs
Use 'sudo apt autoremove' to remove them.
The following packages will be upgraded:
  swtpm-libs
1 to upgrade, 0 to newly install, 0 to remove and 0 not to upgrade.
5 not fully installed or removed.
Need to get 0 B/33.4 kB of archives.
After this operation, 0 B of additional disk space will be used.
(Reading database ... 507644 files and directories currently installed.)
Preparing to unpack .../swtpm-libs_0.6.0-1~jammy1.2_amd64.deb ...
Unpacking swtpm-libs:amd64 (0.6.0-1~jammy1.2) over (0.6.0-1~impish1.2) ...
dpkg: error processing archive /var/cache/apt/archives/swtpm-libs_0.6.0-1~jammy1.2_amd64.deb (--unpack):
 trying to overwrite '/usr/lib/x86_64-linux-gnu/swtpm/libswtpm_libtpms.so.0.0.0', which is also in package swtpm 0.6.3-0ubuntu3
Errors were encountered while processing:
 /var/cache/apt/archives/swtpm-libs_0.6.0-1~jammy1.2_amd64.deb
E: Sub-process /usr/bin/dpkg returned an error code (1)
  [+] Updating /var/cache/deb-get/antimicrox.json
  [+] Updating /var/cache/deb-get/balena-etcher-electron.json
  [+] Updating /var/cache/deb-get/bat.json
  [+] Updating /var/cache/deb-get/bitwarden.json
  [+] Updating /var/cache/deb-get/caprine.json
  [+] Updating /var/cache/deb-get/codium.json
  [+] Updating /var/cache/deb-get/deb-get.json
  [+] deb-get_0.2.6-1_all.deb is up to date.
  [+] discord-0.0.17.deb is up to date.
  [+] Updating /var/cache/deb-get/duf.json
  [+] Updating /var/cache/deb-get/fd.json
  [+] Updating /var/cache/deb-get/figma-linux.json
  [+] Updating /var/cache/deb-get/franz.json
  [+] Updating /var/cache/deb-get/gh.json
  [+] Updating /var/cache/deb-get/git-delta.json
  [+] Updating /var/cache/deb-get/github-desktop.json
  [+] Updating /var/cache/deb-get/grype.json
  [+] Updating /var/cache/deb-get/heroic.json
  [+] heroic_2.2.6_amd64.deb is up to date.
  [+] Updating /var/cache/deb-get/igdm.json
  [+] Updating /var/cache/deb-get/insomnia.json
  [+] Updating /var/cache/deb-get/irccloud-desktop.json
  [+] Updating /var/cache/deb-get/jabref.json
  [+] Updating /var/cache/deb-get/lsd.json
  [+] Updating /var/cache/deb-get/ludo.json
  [+] Updating /var/cache/deb-get/mailspring.json
  [+] Updating /var/cache/deb-get/mattermost-desktop.json
  [+] Updating /var/cache/deb-get/micro.json
  [+] micro-2.0.10-amd64.deb is up to date.
  [+] Updating /var/cache/deb-get/obsidian.json
  [+] Updating /var/cache/deb-get/onlyoffice-desktopeditors.json
  [+] Updating /var/cache/deb-get/pandoc.json
  [+] pandoc-2.18-1-amd64.deb is up to date.
  [+] Updating /var/cache/deb-get/powershell.json
  [+] Updating /var/cache/deb-get/rambox.json
  [+] Updating /var/cache/deb-get/rclone.json
  [+] Updating /var/cache/deb-get/rocketchat.json
  [+] Updating /var/cache/deb-get/rpi-imager.json
  [+] Updating /var/cache/deb-get/sengi.json
  [+] Updating /var/cache/deb-get/simplenote.json
  [+] spotify-client_1.1.84.716.gc5f8b819_amd64.deb is up to date.
  [+] Updating /var/cache/deb-get/syft.json
  [+] Updating /var/cache/deb-get/tidal-hifi.json
  [+] Updating /var/cache/deb-get/trivy.json
  [+] Updating /var/cache/deb-get/whalebird.json
  [+] Updating /var/cache/deb-get/whatsapp-for-linux.json
  [+] Updating /var/cache/deb-get/zenith.json
  [+] zoom_amd64.deb is up to date.
```

**Comment by TuxVinyards** [comment]
swtpm: Could not open UnixIO socket: Permission denied

![Screenshot at 2022-06-11 11-55-22](https://user-images.githubusercontent.com/3956806/173183588-b20d42d5-ac00-4aa6-b8f7-99783e0b57c8.jpg)

Tried forcing a downgrade using Synaptic.  Just goes round in loops. If I downgrade one bit, it throws errors and re-upgrades things somewhere else.

![Screenshot at 2022-06-11 12-28-31](https://user-images.githubusercontent.com/3956806/173184069-30e93448-eb4f-4133-93e5-9bdb66795fad.png)

Also, why am I seeing four versions of quickemu showing.  While, if I uninstall three, it takes out four. 

![Screenshot at 2022-06-11 11-57-26](https://user-images.githubusercontent.com/3956806/173183679-81d9bbf6-9d31-4e6e-bc0d-7af203fd5dad.png)

Terminal output quickemu:
`
- BOOT:     EFI (Windows), OVMF (/usr/share/OVMF/OVMF_CODE_4M.fd), SecureBoot (off).
 - Disk:     windows-11/disk.qcow2 (128G)
             Just created, booting from windows-11/Win11_EnglishInternational_x64v1.iso
 - Boot ISO: windows-11/Win11_EnglishInternational_x64v1.iso
 - CD-ROM:   windows-11/virtio-win.iso
 - Display:  SDL, qxl-vga, GL (on), VirGL (off)
 - ssh:      On host:  ssh user@localhost -p 22220
 - SPICE:    On host:  spicy --title "windows-11" --port 5930 --spice-shared-dir /home/xxx/Public
 - WebDAV:   On guest: dav://localhost:9843/
 - TPM:      windows-11/windows-11.swtpm-sock (21305)
 - 
swtpm: Could not open UnixIO socket: Permission denied

 - Process:  Starting windows-11.conf as windows-11 (21307)
 -
`

All a bit puzzling.

**Comment by TuxVinyards** [comment]
Update:  

I couldn't get  Windows 11 to initiate its build at all.  But then I noticed quickemu (or quickgui?) had built the .conf with secure boot set to 'off' (which is a bit at odds with tpm).  So, once I had set that to 'on'  the Windows CD message with the dot trail appeared ......  and I was able to press a key and set it all going.

I am wondering  if the message about a lack of  UnixIO socket on boot up was just a red herring.   It is still showing but maybe swtpm is just making a complaint that it can't connect to anything, that's all.  And it can't connect because Windows is not booted ...

I entered my standard  Microsoft account login, I was recognized & it all updated. I now have Windows 11 running happily.  The settings panel says the system is 'activated' too.

On the four quickemu entry thing, I am putting this down to Synaptic.  It only happens when I select the 'origins' tab.  Only one entry shows normally. 

Good work  Wimpy  (@flexiondotorg)
